plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -618,7 +619,7 @@ class AzBatchService implements Closeable {
         def pool = getPool(spec.poolId)
         if( !pool ) {
             if( config.batch().canCreatePool() ) {
-                createPool(spec)
+                createPool(spec, task)
```

Comment:
It would make sense to keep the labels in the AzVmPoolSpec object, likely when it's created here

https://github.com/nextflow-io/nextflow/blob/3051cd13da0298eaca142509ac05c7080a1c4328/plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy#L589-L607

Reply:
Makes sense. For the auto pool spec, we could add the metadata to the unique hash so that different resource labels trigger different auto pools. That could lead to many different pools being created, and I don't know whether that would be problematic.